### DISCUSSION TITLE
Changed PR merge settings.

### DISCUSSION THREAD

**Discussion by lovelydinosaur** [mkdocs/mkdocs#3821] [Maintenance]
I've updated the PR merge settings on this repo to *only allow squash merging*.

This is the same policy used on all `github.com/encode` projects, and ensures that project history becomes atomic-by-PR.

<details>
<summary>
Consistently using only squash merging gives a much cleaner history-by-PR.
</summary>

<img width="905" alt="image" src="https://github.com/user-attachments/assets/2f4bc552-3963-44dd-b114-cc0c68c9d422">

---

<img width="1286" alt="image" src="https://github.com/user-attachments/assets/943e4650-ca6e-408c-98bb-8b2a2c53cf8d">
</details>

<details>
<summary>
Compared to the alternative.
</summary>

<img width="901" alt="image" src="https://github.com/user-attachments/assets/76bad6cf-a137-4f6a-945b-8de9f058f48e">

---

<img width="1284" alt="image" src="https://github.com/user-attachments/assets/75ae046f-bb5c-4e5b-bbc9-9e129c764497">
</details>

**Comment by pawamoy** [comment]
Good :+1: Does it still allow the "rebase" option though?

**Reply by pawamoy** [reply]
It doesn't :confused: 

![norebase](https://github.com/user-attachments/assets/2001f4f8-73c9-43fa-97f3-0822b76a4008)

It's not uncommon to have several atomic commits in PRs that shouldn't be squashed.

**Reply by Andre601** [reply]
From how I understand github does this only change the default merge behaviour. Other merge options should still be accessible via the dropdown on the merge button.

Edit: Nevermind that...

**Reply by trel** [reply]
strong rebase proponent here.

squash is second best (limitation being what Andre says about multiple atomic commits remaining separate).

last is merge commits, ew.
